test/test_jit.py

```diff
@@ -9949,6 +9949,14 @@ def foo(a):
         v = torch.rand(10, 3)
         self.assertEqual(torch.chunk(v, dim=0, chunks=2)[0], foo(v))
 
+    def test_trace_chunk_with_list_output(self):
+        def chunk(x):
+            return torch.ops.aten.chunk(x, chunks=2, dim=0)
```

Comment:
It doesn't repro with torch.chunk

Reply:
hmm ok it does not repro with `torch.chunk`, mainly because `torch.chunk` in tracing is handing in VariableType level where it produce `std::vector<at::Tensor>`. 

I think here we should not use the chunk or `torch.ops.aten.chunk` as a test case, because `torch.ops.aten.chunk` is JIT binded custom op that might get changed(although unlikely) in the future. 

What this PR does is basically enable List as the output of the tracer, what we should do here for testing is to add test cases that cover the below use cases:
1. List[Tensor] as trace output, should works and not throw
2. List[other types], should throw to user with the correct error message reporting.
3. nested lists, what should be valid in this case? We should only throw to user if at least one element type is not Tensor. 


Can you add at least these three test cases? For example for case one we should test if the below function works for tracing:

```python
def trace_list_output(x):
    return [x, x, x]
```
similarly for case 2 and 3.

Also, we could just use the `self.checkTrace` helper instead of manual tracing in the test cases. You can refer to https://github.com/pytorch/pytorch/blob/5579611544c6191d147fe4c7d7f5f5def6f44315/test/test_jit.py#L2223 to see how we add those test cases.

Reply:
Good point! Adding tests for these cases. I'll not support nested lists for now, making sure it throws an exception.